src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -86,36 +101,63 @@ protected MQTTBridge(Topics topics, TopicMapping topicMapping, MessageBridge mes
         this.messageBridge = messageBridge;
         this.pubSubClient = new PubSubClient(pubSubIPCAgent);
         this.ioTCoreClient = new IoTCoreClient(iotMqttClient);
-        this.executorService = executorService;
-    }
+        this.mqttClientFactory = () -> new MQTTClient(brokerUri, clientId, mqttClientKeyStore, executorService);
 
-    @Override
-    public void install() {
-        mappingConfigTopics =
-                this.config.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY, MQTT_TOPIC_MAPPING);
+        // handle configuration changes
+        Topics mappingConfigTopics = topics.lookupTopics(BridgeConfig.PATH_MQTT_TOPIC_MAPPING);
+        topics.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY).subscribe((what, child) -> {
```

Comment:
I'm glad you're subscribing to the configuration key now - that avoids/fixes the issue we're seeing in CDA where subscriptions get dropped after a configuration reset. Looks like the same issue was here as well.

Just curious - what's the reason for moving this to the constructor instead of the install method?